code/frameworks/svelte-vite/src/plugins/ts2doc.ts

```diff
@@ -0,0 +1,209 @@
+import ts from 'typescript';
+import { svelte2tsx } from 'svelte2tsx';
+import { VERSION } from 'svelte/compiler';
+import { type SvelteDataItem } from 'sveltedoc-parser';
+
+function getInitializerValue(initializer?: ts.Node) {
+  if (initializer === undefined) {
+    return undefined;
+  }
+  if (ts.isNumericLiteral(initializer)) {
+    return Number(initializer.text);
+  }
+  if (ts.isStringLiteral(initializer)) {
+    return `"${initializer.text}"`;
+  }
+  if (initializer.kind === ts.SyntaxKind.TrueKeyword) {
+    return true;
+  }
+  if (initializer.kind === ts.SyntaxKind.FalseKeyword) {
+    return false;
+  }
+  return initializer.getText();
+}
+
+/*
+ * Make sveltedoc-parser compatible data from .svelte file with svelte2tsx and TypeScript.
+ */
+export function ts2doc(fileContent: string) {
+  const shimFilename = require.resolve('svelte2tsx/svelte-shims-v4.d.ts');
+  const shimContent = ts.sys.readFile(shimFilename) || '';
+  const shimSourceFile = ts.createSourceFile(
+    shimFilename,
+    shimContent,
+    ts.ScriptTarget.Latest,
+    true
+  );
```

Comment:
I'm not sure what this does, could you add a comment explaining these lines? If relevant, feel free to reference how this is used internally in Svelte too, or docs.

Reply:
Actually, I’m not entirely sure myself. I just referenced these lines from the svelte2tsx code: https://github.com/sveltejs/language-tools/blob/8d3a335f70cccdde599b8bc69b1830c0af478298/packages/svelte2tsx/src/emitDts.ts#L101

Reply:
Removed `typescript` dependency.